sync/src/synchronizer/mod.rs

```diff
@@ -167,15 +167,15 @@ impl<CS: ChainStore> Synchronizer<CS> {
     }
 
     //TODO: process block which we don't request
-    pub fn process_new_block(&self, peer: PeerIndex, block: Block) {
+    pub fn process_new_block(&self, peer: PeerIndex, block: Block) -> Result<(), FailureError> {
         if self.orphan_block_pool.contains(&block) {
             debug!("block {:x} already in orphan pool", block.header().hash());
-            return;
+            return Ok(());
         }
 
         match self.shared().get_block_status(&block.header().hash()) {
             BlockStatus::VALID_MASK => {
-                self.insert_new_block(peer, block);
+                self.insert_new_block(peer, block)?;
             }
             status => {
```

Comment:
Why not return an error when `BlockStatus` is unexpected?

Reply:
I'm not sure it's a case to ban this peer

Reply:
I am not sure, either. Keep as it is.